### DISCUSSION TITLE
Use an install command in the package.json

### DISCUSSION THREAD

**Discussion by venomdev** [npm/feedback#891] [Registry]
This might already exist but I'm not finding any clear documentation on it.

I've noticed that the install command is generally `npm i {package-name}` where in some instances it should be `npm i --save-dev {package-name}` 

Is there a property of the package.json that we can set that appears on the npm site for the install command? If not could it be added so that we can set that up for custom installation commands?

I would also like to extend the command to include the main package that my package is a plugin for. So it would actually say something like this:

`npm i --save-dev gulp@4 my-gulp-plugin`

Then the user can use the copy button and have a working install with a simple copy and paste.

**Comment by ljharb** [comment]
No, because it's impossible for a package author to know if the package should actually be a prod dep or a dev dep, or even a peer dep. Nothing's always one of them.

**Comment by venomdev** [comment]
Well it's now always a prod dep so that is a forced choice anyway. 

If someone knows they need a peer dep they probably won't do a simple copy and paste anyway. The normal prod dep would still fail in a peer dep case wouldn't it?

**Reply by ljharb** [reply]
That the default is a prod dep isn't really the point; if you know you need to override the default when installing something, you will.

(ftr, for a gulp plugin, _another plugin_ would need to peer-depend on your plugin, not dev-depend on it, to have things work properly)

**Reply by venomdev** [reply]
"if you know you need to override the default when installing something, you will."

With the current setup up you have to.

If you don't have gulp installed as a devDependencies then it won't matter if you have gulp-rename installed. It doesn't make it a peer dependency if you aren't using gulp as a devDependency in the first place. The gulp plugin won't work on it's own.

**Reply by ljharb** [reply]
I understand that a default of "prod" won't always be correct. However, a default of "dev" would risk breaking things, which is worse.

Since there's no way to programmatically know what the appropriate kind of dep is for a given use case, it's always going to be up to the human to choose - and the safest default is a prod dep.

**Reply by venomdev** [reply]
"the human to choose - and the safest default"

These are complete opposites... a default and a choice mean different things.

I can safely say in 99% of cases that gulp and grunt aren't dependencies. I'm almost 100% certain that webpack is never a dependency.

**Reply by venomdev** [reply]
As a npm package developer our job is to make other developers jobs easier. Those with special edge cases will know when to modify the command.

**Reply by ljharb** [reply]
Right - and webpack, gulp, and grunt are special edge cases. 99% of the time people are installing prod deps, not dev deps or peer deps.
